x-pack/plugins/security_solution/common/experimental_features.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export type ExperimentalFeatures = typeof allowedExperimentalValues;
+
+/**
+ * A list of allowed values that can be used in `xpack.securitySolution.enableExperimental`.
+ * This object is then used to validate and parse the value entered.
+ */
+const allowedExperimentalValues = Object.freeze({
+  fleetServerEnabled: false,
+});
+
+type Mutable<T> = { -readonly [P in keyof T]: T[P] };
+
+const SecuritySolutionInvalidExperimentalValueError = class extends Error {};
+
+/**
+ * A Map of allowed experimental values in uppercase (for data normalization purposes) to an object
+ * containing internal information about the value (currently only the non-uppercased key).
+ * Used for validation of values entered via the plugin config setting.
+ */
+const allowedKeys = (Object.keys(allowedExperimentalValues) as Array<
+  keyof ExperimentalFeatures
+>).reduce((map, key) => {
+  map.set(key.toUpperCase(), { key });
+  return map;
+}, new Map<string, { key: keyof ExperimentalFeatures }>());
+
+/**
+ * Parses the string value used in `xpack.securitySolution.enableExperimental` kibana configuration,
+ * which should be a string of values delimited by a comma (`,`)
```

Comment:
why this way? Can it be an array? that would skip the parsing.

Can it be an object? Then the keys could have values?

I guess I don't understand why the value of `enableExperimental` can be multiple things, instead of just `true`/`false` for things we're calling pre-beta features.

e.g., 

```yaml
xpack.securitySolution.enableExperimental: "fleetServerEnabled,transformlessFuture,paulHoodUp"
```
that seems to be the proposal here?

The string parsing makes me go :thinking:  And want

```yaml
xpack.securitySolution.enableExperimental: ['fleetServerEnabled', 'transformlessFuture', 'paulHoodUp']
```
Or long-form syntax:

```yaml
xpack.securitySolution.enableExperimental:
  - fleetServerEnabled
  - transformlessFuture
  - paulHoodUp
```

And then the thought is, ok, what if one of those things requires a value, then you can make it an object:

```yaml
xpack.securitySolution.enableExperimental:
  fleetServerEnabled: true
  transformlessFuture: OMG
  paulHoodUp: yes
```

and at that point we're sort of back to where we currently are.. just nested under the `enableExperimental` key.

Reply:
Thanks @pzl for the feedback. I like the array idea as that would be a much cleaner approach. I'm not sure about the object.

The goal is to have 1 single setting that can control `n` number of experimental stuff that we don't want exposed just yet. This 1 property will then be whitelisted (see my other PR over at `cloud` project) for cloud so that we can start up dev environments  and turn these things on and not have to keep introducing new configuration options and then go through the process of whitelisting them.

I'll look at the cloud definition for whitelisting again, but I'm thinking that setting it to an Object would require us to define the know "keys" upfront, which I'm trying to avoid. I think the Array setting is possible and would scale better.